### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by khinkhinn**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -322,6 +336,20 @@ The diagram below shows a sample interaction of `TaskDoneCommand`.
...
+The TaskDelete command allows the user to delete a task from a tag. 
+
+1. The command is passed into `LogicManager`.
+2. `LogicManager` calls the parseCommand method of `ProjactParser`.
+3. `ProjactParser` identifies the commandWord, which in this case is `taskdone` and the arguments.
+4. `ProjactParser` calls the parse method of `TaskDeleteCommandParser`, which parses the argument, cand returns a new `TaskDeleteCommand`.
+5. The `LogicManager` then calls the execute method of the `TaskDeleteCommand`, which calls the setTag method of `Model`.
+
+The diagram below shows a sample interaction of `TaskDeleteCommand`. 
+
```
Comment (yeo-kexin) [inline docs/DeveloperGuide.md:354]: do you want to include why you implemented it this way? e.g. alphabetical index

File: docs/diagrams/TaskDeleteSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+TaskDeleteCommand --> TaskDeleteCommandParser : Command
+deactivate TaskDeleteCommand
+
+TaskDeleteCommandParser --> ProjactParser : Command
+deactivate TaskDeleteCommandParser
+TaskDeleteCommandParser -[hidden]-> ProjactParser
+destroy TaskDeleteCommandParser
+
+ProjactParser --> LogicManager : command
```
Comment (yeo-kexin) [inline docs/diagrams/TaskDeleteSequenceDiagram.puml]: can standardise capitalisation! same issue for tagadd sequence diagram

**Review by yeo-kexin — COMMENTED** [review]
lgtm apart from some comments

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (siqiang-ng) [reply docs/DeveloperGuide.md:354]: Ive edited the changes as this comment affects TaskDone as well!

**Comment by siqiang-ng** [comment]
lgtm

**Review by siqiang-ng — APPROVED** [review]

**Pull request merged**